pyccel/parser/semantic.py

```diff
@@ -344,11 +344,22 @@ def change_to_module_scope(self):
 
     def check_for_variable(self, name):
         """
+        Search for a Variable object with the given name in the current scope.
+
         Search for a Variable object with the given name in the current scope,
         defined by the local and global Python scopes. Return None if not found.
-        """
 
-        if self.current_class:
+        Parameters
+        ----------
+        name : str
+            The object describing the variable.
+
+        Returns
+        -------
+        Variable
+            Returns the varibale if found or None.
+        """
+        if self.current_class and not name == 'self':
             for i in self._current_class.attributes:
                 if i.name == name:
                     var = i
```

Comment:
Should this bit call `check_for_attribute`?
Do we need both `check_for_variable` and `check_for_attribute`?
Or could we do:
```python
def check_for_variable(self, name):
    if isinstance(name, DottedName):
        # body of check_for_attribute
    else:
        # body of check_for_variable
```
?